static/js/subs.js

```diff
@@ -167,6 +170,24 @@ function update_stream_pin(sub, value) {
     sub.pin_to_top = value;
 }
 
+function update_stream_mandatory_status(sub, value) {
+    sub.mandatory_stream = value;
+    var mandatory_button_selc = $('.toggle-stream-mandatory[data-stream-name="' + sub.name + '"]');
+    var new_button_text;
+
+    // Change button's text when mandatory status is updated
+    if (value) {
+        new_button_text = "Change stream mandatory status to false";
```

Comment:
This phrasing is a bit awkward. I'd change it to something like 'Make stream optional'